mlflow/pyfunc/model.py

```diff
@@ -228,17 +231,18 @@ def _save_model_with_class_artifacts_params(
     if artifacts:
         saved_artifacts_config = {}
         with TempDir() as tmp_artifacts_dir:
-            tmp_artifacts_config = {}
             saved_artifacts_dir_subpath = "artifacts"
             for artifact_name, artifact_uri in artifacts.items():
-                tmp_artifact_path = _download_artifact_from_uri(
-                    artifact_uri=artifact_uri, output_path=tmp_artifacts_dir.path()
-                )
-                tmp_artifacts_config[artifact_name] = tmp_artifact_path
-                saved_artifact_subpath = posixpath.join(
-                    saved_artifacts_dir_subpath,
-                    os.path.relpath(path=tmp_artifact_path, start=tmp_artifacts_dir.path()),
-                )
+                if artifact_name == "snapshot":
```

Comment:
Does this approach work if the logging and loading happen in different machines?

Reply:
When users use 'snapshot_download' to download the model, local_dir is the place to store it. And transformers 'from_pretrained' load from the path, the path should be a local directory. 
If loading happens on a different machine, I don't think transformers works, but they can always have workaround like uploading the model to a storage account the both machines can have access to...

Reply:
@harupy Haru are you talking about the multi-machine cluster scenario? My understanding is in cluster setup, model instantiation/download/save will always happen in the primary machine, the later on pass to workers based on the coordination strategy. So we should be good here.